### PR TITLE
update package to work on 0.6 and use StaticArrays

### PR DESCRIPTION
**Pull request by getzdan**

### CODE REVIEW

**Comment by mschauer** [comment]
Thank you so much! Can mark the option "Allow edits from maintainers" on the right side if you have this?

>If checked, users with write access to mschauer/Bridge.jl/ can add new commits to your branch. You can always change this setting later.

**Comment by getzdan** [comment]
"Allow edits" option already marked, and edits are welcome.

**Comment by mschauer** [comment]
the macro isdefined is not yet defined under julia 0.6..., I define a macro _isdefined like in 
https://discourse.julialang.org/t/check-whether-a-variable-is-defined/1018/2 in misc
to circumvent

File: src/euler.jl [hunk 1]
```diff
@@ -13,7 +13,9 @@ function euler!(Y, u, W::SamplePath, P)
 
     for i in 1:N-1
         yy[.., i] = y
-        y = y + b(tt[i], y, P)*(tt[i+1]-tt[i]) + σ(tt[i], y, P)*(ww[.., i+1]-ww[..,i])
+        factor = σ(tt[i], y, P)
+        isa(factor,UniformScaling) && ( factor = factor.λ )
+        y = y + b(tt[i], y, P)*(tt[i+1]-tt[i]) + factor*(ww[.., i+1]-ww[..,i])
     end
```
Comment (mschauer) [inline src/euler.jl]: Can you tell me when multiplying with UniformScaling directly is a problem?

Comment (getzdan) [reply src/euler.jl]: The problem was not multiplying by a UniformScaling, it was adding to the other terms. Can `ww[.., i+1]-ww[..,i]` be some type like SDiagonal. I'll check.

File: test/guip.jl [hunk 2]
```diff
@@ -1,11 +1,11 @@
-using Bridge, FixedSizeArrays, Distributions
+using Bridge, StaticArrays, Distributions
 using Base.Test
 #import Bridge: b, σ, a, transitionprob
-const percentile = 2.3 
+const percentile = 100.3
```
Comment (mschauer) [inline test/guip.jl]: what would be a good procedure for probabilistic tests? would it be ok for the test to fail with a small probability?

File: src/cspline.jl [hunk 3]
```diff
@@ -1,10 +1,11 @@
...
+using Compat, StaticArrays
 function cspline(s, t1, t2, p1, p2, m1, m2)
+    # @show (s, t1, t2, p1, p2, m1, m2)
     d = t2-t1
     t = (s-t1)/(t2-t1)
     t2 = t*t
     t3 = t2*t
-    z = @fsa([2. -3. 0. 1.; -2. 3. 0. 0.; 1. -2. 1. 0.; 1. -1. 0. 0.])* @fsa([t3, t2, t, 1.])
+    z = (@SMatrix [2. -3. 0. 1.; -2. 3. 0. 0.; 1. -2. 1. 0.; 1. -1. 0. 0.])* (@SVector [t3, t2, t, 1.])
     z[1]*p1 + z[2]*p2 + z[3]*d*m1 + z[4]*d*m2
```
Comment (mschauer) [inline src/cspline.jl]: i have unrolled this, updating later

File: test/guip.jl (hunk 2 above)
Comment (getzdan) [reply test/guip.jl]: good question. perhaps `srand(.)` is the way to go. maybe to be creative `srand(package_version)`.

File: example/fitzhugh_nagumo.jl [hunk 4]
```diff
@@ -167,17 +128,18 @@ if false
     plot(Yobs[1:2],"o:")
 
     i = 4
-    v = Vec(Matrix(L)\Vector(Yobs.yy[i+1]))
+    v = SVector(Matrix(L)\Vector(Yobs.yy[i+1]))
     P° = PBridgeProp(Ptrue, Ytrue[i]..., Yobs.tt[i+1], v, Ytrue[i+2]..., L,si^2*I, Ptrue.a)
-    B = eulerb(sample(tt[m*(i-1)+1:m*(i+1)+1] , Wiener{Vec{2,Float64}}()),P°)
+    B = bridge(sample(tt[m*(i-1)+1:m*(i+1)+1] , Wiener{SVector{2,Float64}}()),P°)
```
Comment (mschauer) [inline example/fitzhugh_nagumo.jl:135]: Here I am not sure yet what is the right way to factor the code of euler methods and euler methods where the last point is singular and therefore taken from a fixed value

File: example/REQUIRE [hunk 5]
```diff
@@ -0,0 +1 @@
+ConjugatePriors
```
Comment (mschauer) [inline example/REQUIRE:1]: This kind of sub-directory dependency is for example evaluated for test directories

**Comment by mschauer** [comment]
Green light by travis on 0.6

File: example/REQUIRE (hunk 5 above)
Comment (getzdan) [reply example/REQUIRE:1]: Good to know. The examples IIRC also do some plotting. What will be the stance on plotting? Will it have to be explicitly enabled?
The REQUIRE is in `example` directory, shouldn't it be in the `test` directory?

Comment (mschauer) [reply example/REQUIRE:1]: It is not needed for the tests. This is just an informal way to state that the examples have additional requirements on top of REQUIRE. We should put this into the readme too.

Comment (mschauer) [reply example/REQUIRE:1]: Do you have preferred stance?

**Comment by getzdan** [comment]
Question: If the package is going to go into METADATA, should it be renamed to Bridges.jl ? (Bridge.jl is also fine)

**Comment by mschauer** [comment]
Haha, that is the most difficult question. You prefer Bridges?

**Comment by getzdan** [comment]
Not sure. Bridge.jl rolls better on the tongue, but Bridges.jl is the plural standard (perhaps also RandomBridges or BrownianBridges are good names). Currently the bridges are actually Brownian, is there an intent to expand into other stochastic process bridges?

**Comment by mschauer** [comment]
General Diffusion process bridge-approximations, Gamma process bridges and Bessel process bridges are included already

**Comment by getzdan** [comment]
OK. Then it's up to you. No preference here.

**Comment by mschauer** [comment]
Lets keep it, I also like the rhythm. (Officially because of "A less systematic name may suit a package that implements one of several possible approaches to its domain" and there are quite some ways of generate Brownian motions path for example already). Can you check the multiplication with the scaling matrix?

**Comment by getzdan** [comment]
The problem with multiplication with scaling is that sometimes the other multiplicand is a scalar and then the result is a UniformScaling. This is then *added* to another scalar which causes an error (or warning) in 0.6 or 0.7 (search the release notes https://github.com/JuliaLang/julia/blob/master/NEWS.md for UniformScaling). This is the reason why this line is there, but perhaps I was too hasty and fixed the wrong problem.

**Comment by mschauer** [comment]
we can have a _scale function catching the problem for now

**Pull request merged**